### DISCUSSION TITLE
DynamoDB test initializations

### DISCUSSION THREAD

**Discussion by nicklasweasel** [quarkusio/quarkus#35951] [Community]
Hi,

  How do people generally go about the task of e.g. creating a DynamoDB table against a localstack before tests and deleting it afterwards?

Apparently QuarkusTestResourceLifecycleManager implementations aren't CDI aware so I can't inject a DynamoDbEnhancedClient. And Arc.container().instance(DynamoDbEnhancedClient.class).get() gives null for the container so it doesn't look enabled.

I could of course create the client->enhanced client through builders but apparently application.properties aren't read because it gets created somewhere else (my default AWS credentials?) because I get "table not found" when running the test.

I could of course force the builder with the same stuff as in the application.properties but at that point I came to the conclusion that "it can probably be done in a more sane way" and that's why I'm here ;-)

Thanks in advance,
  Nik

**Comment by axelmyrberg** [comment]
We inject the dynamoDbenhancedClient in the test class and call `dynamoDbenhancedClient.table(getTableName(), TableSchema.fromClass(YourEntity.class)).createTable()` in the junit BeforeEach to create the table before tests.

**Reply by nicklasweasel** [reply]
Thanks for the reply. Yes, it was pretty much what I ended up with. And most test-lifecycle-actions are apparently better done with junit @Before*/@After*

**Comment by trico-tillster** [comment]
This can also be achieved with https://github.com/quarkiverse/quarkus-amazon-services/pull/317
